source/Conversation.cpp

```diff
@@ -402,40 +411,76 @@ int Conversation::NextNode(int node, int choice) const
 			|| static_cast<unsigned>(choice) >= nodes[node].data.size())
 		return DECLINE;
 	
-	return nodes[node].data[choice].second;
+	return nodes[node].data[choice].next;
+}
+
+
+
+// Return whether the text has failed conditions and should be skipped
+bool Conversation::ShouldSkipText(const map<string, int64_t> &vars, int node, int choice) const
+{
+	if(static_cast<unsigned>(node) >= nodes.size() || static_cast<unsigned>(choice) >= nodes[node].data.size())
```

Comment:
This bounds check should probably be pulled out into a private method

Reply:
There are lots of bounds checks that are duplicated inline like that. Should I fix all of them?

Reply:
I hope 33b87e3a is satisfactory. I'll leave it to you to hit "resolve" if so.